portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -82,36 +84,65 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery commentHistory = datastore.prepare(commentHistoryQuery);
     List<Entity> commentHistoryList = commentHistory.asList(FetchOptions.Builder.withDefaults());
     int totalComments = commentHistoryList.size();
-    int totalPages;
 
     // Obtain the maximum number of comments to display. If the number is invalid, take the default value.
```

Comment:
This function is getting a bit long. Would you be able to break each of these logical sections up into helper functions (with relevant names)?

Reply:
Yes, I'm working on it!